From e727991a269ea8a902ad0f3692662a1899456eae Mon Sep 17 00:00:00 2001
From: David S. Wang <dsw@cloudera.com>
Date: Wed, 27 Mar 2013 16:39:07 -0700
Subject: [PATCH 40/96] HBASE-7657 Make ModifyTableHandler synchronous

Reason: Bug
Author: Himanshu Vashishtha
Ref: CDH-10057
(cherry picked from commit 3f00f4d27818ffdb4e7d661849779291199eb218)
---
 .../org/apache/hadoop/hbase/client/TestAdmin.java  |   29 +------------------
 1 files changed, 2 insertions(+), 27 deletions(-)

diff --git a/src/test/java/org/apache/hadoop/hbase/client/TestAdmin.java b/src/test/java/org/apache/hadoop/hbase/client/TestAdmin.java
index 66b5dd7..dbf8644 100644
--- a/src/test/java/org/apache/hadoop/hbase/client/TestAdmin.java
+++ b/src/test/java/org/apache/hadoop/hbase/client/TestAdmin.java
@@ -377,7 +377,7 @@ public class TestAdmin {
     copy.setValue(key, key);
     boolean expectedException = false;
     try {
-      modifyTable(tableName, copy);
+      admin.modifyTable(tableName, copy);
     } catch (TableNotDisabledException re) {
       expectedException = true;
     }
@@ -468,7 +468,7 @@ public class TestAdmin {
     copy.setValue(key, key);
     boolean expectedException = false;
     try {
-      modifyTable(tableName, copy);
+      admin.modifyTable(tableName, copy);
     } catch (TableNotDisabledException re) {
       expectedException = true;
     }
@@ -476,31 +476,6 @@ public class TestAdmin {
   }
 
   /**
-   * Modify table is async so wait on completion of the table operation in master.
-   * @param tableName
-   * @param htd
-   * @throws IOException
-   */
-  private void modifyTable(final byte [] tableName, final HTableDescriptor htd)
-  throws IOException {
-    MasterServices services = TEST_UTIL.getMiniHBaseCluster().getMaster();
-    ExecutorService executor = services.getExecutorService();
-    AtomicBoolean done = new AtomicBoolean(false);
-    executor.registerListener(EventType.C_M_MODIFY_TABLE, new DoneListener(done));
-    this.admin.modifyTable(tableName, htd);
-    while (!done.get()) {
-      synchronized (done) {
-        try {
-          done.wait(100);
-        } catch (InterruptedException e) {
-          e.printStackTrace();
-        }
-      }
-    }
-    executor.unregisterListener(EventType.C_M_MODIFY_TABLE);
-  }
-
-  /**
    * Listens for when an event is done in Master.
    */
   static class DoneListener implements EventHandler.EventHandlerListener {
-- 
1.7.0.4

